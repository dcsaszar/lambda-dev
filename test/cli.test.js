const CliTest = require('command-line-test');

describe('cli', () => {
  test('Without arguments', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });

  test('Build without entry argument', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js build');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });

  test('Build without target argument', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js build test/fixtures');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });

  test('Serve without entry argument', async () => {
    const cli = new CliTest();
    const { error, stdout, stderr } = await cli.exec('node cli.js serve');

    expect(error).toMatchSnapshot();
    expect(stdout).toEqual(null);
    expect(stderr).toEqual(null);
  });
});
