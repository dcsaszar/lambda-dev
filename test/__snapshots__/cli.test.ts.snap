// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli Build without entry argument 1`] = `
[Error: Command failed: ts-node ./src/cli.ts build

/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: тип Unable to compile TypeScript:
src/cli.ts(35,12): error TS2304: Cannot find name 'CliServeArgs'.
src/cli.ts(57,12): error TS2304: Cannot find name 'CliBuildArgs'.

    at createTSError (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261:12)
    at getOutput (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:367:40)
    at Object.compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:558:11)
    at Module.m._compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:439:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:442:12)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
]
`;

exports[`cli Build without target argument 1`] = `
[Error: Command failed: ts-node ./src/cli.ts build test/fixtures

/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: тип Unable to compile TypeScript:
src/cli.ts(35,12): error TS2304: Cannot find name 'CliServeArgs'.
src/cli.ts(57,12): error TS2304: Cannot find name 'CliBuildArgs'.

    at createTSError (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261:12)
    at getOutput (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:367:40)
    at Object.compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:558:11)
    at Module.m._compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:439:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:442:12)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
]
`;

exports[`cli Serve without entry argument 1`] = `
[Error: Command failed: ts-node ./src/cli.ts serve

/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: тип Unable to compile TypeScript:
src/cli.ts(35,12): error TS2304: Cannot find name 'CliServeArgs'.
src/cli.ts(57,12): error TS2304: Cannot find name 'CliBuildArgs'.

    at createTSError (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261:12)
    at getOutput (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:367:40)
    at Object.compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:558:11)
    at Module.m._compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:439:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:442:12)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
]
`;

exports[`cli Without arguments 1`] = `
[Error: Command failed: ts-node ./src/cli.ts is

/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261
    return new TSError(diagnosticText, diagnosticCodes)
           ^
TSError: тип Unable to compile TypeScript:
src/cli.ts(35,12): error TS2304: Cannot find name 'CliServeArgs'.
src/cli.ts(57,12): error TS2304: Cannot find name 'CliBuildArgs'.

    at createTSError (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:261:12)
    at getOutput (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:367:40)
    at Object.compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:558:11)
    at Module.m._compile (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:439:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:700:10)
    at Object.require.extensions.(anonymous function) [as .ts] (/Users/iiro/git/lambda-dev/node_modules/ts-node/src/index.ts:442:12)
    at Module.load (internal/modules/cjs/loader.js:599:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:538:12)
    at Function.Module._load (internal/modules/cjs/loader.js:530:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:742:12)
]
`;
